app/javascript/packs/webauthn-authenticate.ts

```diff
@@ -36,6 +37,10 @@ function webauthn() {
         (document.getElementById('signature') as HTMLInputElement).value = result.signature;
         webauthnInProgressContainer.classList.add('display-none');
         webauthnSuccessContainer.classList.remove('display-none');
+        // Check if alert container is shown and remove when device passes successfully.
+        if (webauthAlertContainer) {
+          webauthAlertContainer.remove();
+        }
```

Comment:
Technically we're not checking if it's shown, since the element will always exist in the markup, and we're removing it whether or not it's visible. To that point, we may be able to even go as far to assume the element exists and assign the variable above using the [TypeScript non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator), similar to what we're doing with `webauthnInProgressContainer` and `webauthnSuccessContainer` variables before it.

Reply:
I had encountered unwanted behavior where the user not clicking cancel initially got an error then had to re-attempt in order to authenticate. I could not discern why but adding the check with that if statement fixed it.
The element with the class .usa-alert--error gets inserted as that view loads and if it needs to display the error. If there's no error that element doesn't show up in the DOM.
Would it be more preferable to use a non-null operator?